pandas/tests/indexes/multi/test_reindex.py

```diff
@@ -101,3 +101,13 @@ def test_reindex_non_unique():
     msg = "cannot handle a non-unique multi-index!"
     with pytest.raises(ValueError, match=msg):
         a.reindex(new_idx)
+
+
+def test_reindex_fill_added_rows():
+    # GH: 23693
+    i = pd.MultiIndex.from_tuples([("a", "b"), ("d", "e")])
+    df = pd.DataFrame([[0, 7], [3, 4]], index=i, columns=["x", "y"])
+    i2 = pd.MultiIndex.from_tuples([("a", "b"), ("d", "e"), ("h", "i")])
+    result = df.reindex(i2, axis=0, method="ffill")
+    expected = pd.DataFrame([[0, 7], [3, 4], [3, 4]], index=i2, columns=["x", "y"])
```

Comment:
if the method being tested is DataFrame.reindex, then this belongs in tests.frame.method.test_reindex

Reply:
Sorry, was checking frame/indexing and could not find a reindex file. Moved the test now to the methods folder